test/Microsoft.ML.Benchmarks/Program.cs

```diff
@@ -29,7 +29,7 @@ static void Main(string[] args)
 
         private static IConfig CreateCustomConfig() 
             => DefaultConfig.Instance
-                .With(Job.Default
+                .With(Job.VeryLongRun
```

Comment:
Do we really want to change this?  /cc @adamsitnik

Reply:
I think that we should not change it.

What the current setting is : 1 warmup iteration, up to 20 workload iterations in 1 process
What LongRun is : up to 30 warmup iterations, up to 500 workload iterations repeated in 4 dedicated processes

Reply:
What's the effect of having warm-up iterations?

Are we running in the same process, or another? The normal user has a single run, so I think only the first warm-up iteration is representative. For instance, the WordEmbedding transform only spends time loading the model the first time; this time is significant (seconds to minutes). I'm not sure if the static loading of the word embedding models will be torn-down in the current tests. 

Static dictionary which keeps the word embedding models:
https://github.com/dotnet/machinelearning/blob/b727d10c02a8f093b72ff98a02e2d63f1051d39a/src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs#L84

If we run benchmarks in concurrency, we'll hit the loader lock, which will cause only one model to load per process in parallel:
https://github.com/dotnet/machinelearning/blob/b727d10c02a8f093b72ff98a02e2d63f1051d39a/src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs#L80

Can we check, for the WordEmbedding transforms benchmark, if the model is reloaded from disk for iteration (it should be)?

Reply:
we should be able to verify from the time of warmup iterations. There shouldn't be large difference between both the cases

Reply:
@justinormont every benchmark is executed in a dedicated process. We have 1 warmup iteration to exclude the cost of Jitting from the final result.

However, if in the real life scenarios this code is executed only once it makes sense to have a custom config for it, which is going to spawn new process `n` times and every process is going to run the benchmark exactly once. This is possible with BenchmarkDotNet.

@justinormont could you say something more about typical scenarios? (how many times train and predict are executed)

@davidwrighton this could be also interesting from the Runtime perspective, especially with tiered jitting enabled (cc @kouvel)

Reply:
There isn't just one type of user. What's being tested here is a batch scoring scenario where the user has a pre-trained model and they want to score a dataset against the model. This benchmark is meant to cover the speed of this scoring process (it doesn't return a useful accuracy metric as there is data leakage from running in a Train-Train (mostly same dataset used for both training and the final eval). The user in this benchmark likely is expected to have a cold-start.

Todo: see if the subsequent iterations also load the Word Embedding file (or we won't be including its load-time in the benchmark).

Future pull requests: We should expand the user scenarios tested. And increase the datasets so we don't overfit our perf improvements to only the few represented datasets.